--- xdcc.tcl.old	2017-06-29 15:22:02.309957172 -0400
+++ xdcc.tcl	2017-06-29 15:22:27.669623737 -0400
@@ -3270,6 +3270,10 @@
   } elseif {[string equal $arg "-maxthreads"]} {
     incr i
     append other_syslink_switches " -maxthreads " [lindex $argv $i]
+  } elseif {[string equal $arg "-maxjobs"]} {
+# HG: Modification for IC grid
+    incr i
+    append other_syslink_switches " -maxjobs " [lindex $argv $i]
   } elseif {[string equal $arg "-sdcard"]} {
     incr i
     set tempdir [lindex $argv $i]
